api/rtx/src/unsupported_malloc.c

```diff
@@ -0,0 +1,162 @@
+    heap_size -= index_size;
+
+    /* The box index is at the beginning of the bss section */
+    RtxBoxIndex *const indexOS = box_bss;
+    /* Zero the _entire_ index, so that we know to initialize the mutex on
+     * first use! */
+    memset(box_bss, 0, index_size);
+    /* Initialize user context */
+    indexOS->index.ctx = NULL;
+    /* Initialize box heap */
+    indexOS->index.box_heap = box_bss + index_size;
+    indexOS->index.box_heap_size = heap_size;
+    /* Active heap pointer is NULL */
+    indexOS->index.active_heap = NULL;
+
+    /* There is no box config for unsupported! */
+    indexOS->index.config = NULL;
+
+    /* Set the index */
+    __uvisor_ps = indexOS;
+}
+
+/* uVisor hook for unsupported platforms */
+void secure_malloc_init(void)
+{
+    /* get the main heap size from the linker script */
+    uint32_t heap_size = ((uint32_t) __HeapLimit -
+                          (uint32_t) __end__);
+    /* Main heap size is aligned to page boundaries n*UVISOR_PAGE_SIZE */
+    uint32_t heap_start = (uint32_t) __StackLimit - heap_size;
+    /* align the start address of the main heap to a page boundary */
+    heap_start &= ~(UVISOR_PAGE_SIZE - 1);
+    /* adjust the heap size to the new heap start address */
+    heap_size = (uint32_t) __StackLimit - heap_start;
+
+    /* page heap now extends from the previous main heap start address
+     * to the new main heap start address */
+    extern uint32_t __uvisor_page_size;
+    page_allocator_init(__end__, (void *) heap_start, &__uvisor_page_size);
+    box_index_init((void *) heap_start, heap_size);
+}
+
+
+int uvisor_page_malloc(UvisorPageTable *const table)
+{
+    return page_allocator_malloc(table);
+}
+
+int uvisor_page_free(const UvisorPageTable *const table)
+{
+    return page_allocator_free(table);
+}
+
+/* This is the fallback implementation for using the page allocator from uVisor
+ * inside an OS as a normal function.
+ * Be aware that the page allocator is not re-entrant, so the OS must provide a
+ * mutex implementation to enable thread-safety!
+ */
+#define DPRINTF(...) {}
+#define g_active_box 0
+#define vmpu_is_box_id_valid(...) 0
+#define vmpu_public_flash_addr(...) 1
+#define vmpu_sram_addr(...) 1
+#define HALT_ERROR(id, ...) {}
+
+/* Implement mutex for page allocator. */
+static osMutexId g_page_allocator_mutex_id = NULL;
+static int32_t g_page_allocator_mutex_data[4];
+static const osMutexDef_t g_page_allocator_mutex = { g_page_allocator_mutex_data };
+
+static void page_allocator_mutex_aquire()
+{
+    if (g_page_allocator_mutex_id == NULL) {
+        /* Create mutex if not already done. */
+        g_page_allocator_mutex_id = osMutexCreate(&g_page_allocator_mutex);
+        if (g_page_allocator_mutex_id == NULL) {
+            /* Mutex failed to be created. */
+            return;
+        }
+    }
+
+    osMutexWait(g_page_allocator_mutex_id, osWaitForever);
+}
+
+#define UVISOR_PAGE_ALLOCATOR_MUTEX_AQUIRE  page_allocator_mutex_aquire()
```

Comment:
Maybe I should move this stuff into it's own file `unsupported_page_allocator.c`?